### ISSUE TITLE
Add support for sending files as raw HTTP bodies

### ISSUE THREAD

**Issue by edeckers** [edeckers/react-native-blob-courier#137]
Originally suggested by @pke in #135

Currently `react-native-blob-courier` supports multipart file uploads only, but there are situations where it is desirable to send a file to the server as a raw HTTP body.

**Proposed solution**
Add a `send(request:SendRequest)` method to TypeScript code base, inspired by [HttpClient.Send](https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.send?view=net-5.0).

`SendRequest` would look something like:
```tsx
interface SendRequest {
  method:string;
  absolutePath:string;
  headers:dictionary;
  returnResponse: boolean;
}
```

**Progress**
- [ ] Add `send(request:SenRequest)` method to TypeScript code base
- [ ] Add a `send(method:string,absolutePath:string,headers:dictionary)` method to Android code base
- [ ] Add a `send(method:string,absolutePath:string,headers:dictionary)` method to iOS code base
- [ ] Add TypeScript tests
- [ ] Add Android tests
- [ ] Add iOS tests

**Comment by pke** [comment]
Why not use fetch's RequestInit as input? it contains all the fields already and could be extended to add fileURI as a new prop.
Remember headers in the true HTTP protocol sense is not just a dictionary because there could be multiple named headers.

**Comment by edeckers** [comment]
Good catch, and you're absolutely right, however: for the request interfaces I already take inspiration from `fetch` and `fetch` does actually treat headers as a dictionary and combines duplicate headers into a comma separated list[[1]](https://developer.mozilla.org/en-US/docs/Web/API/Headers),[[2]](https://stackoverflow.com/q/63204093), so I'm sticking with that.

Using `RequestInit` is a good suggestion as well, but it supports fields that aren't supported by the native code base _yet_ or even _ever_. So that would communicate false expectations to users of the library.

**Comment by hatem-72** [comment]
Hello !

Any progress on this feature ?

I am currently looking for a light/efficient library to upload images in Google Cloud Storage using a signed URL. For some reasons using `BlobCourier.uploadBlob` succeed (from an http point of view) but does not upload anything concrete to the bucket. On the other hand calling `ReactNativeBlobUtil.fetch` from `react-native-blob-util` with the same path, url and headers works correctly. I suspect that the `multipart upload` plays a role in this difference in behaviour.

In short I'm not sure my case is relevant but I would be enthusiastic to use the clear and concise syntax of this library and especially to get rid of the `rn-fetch-blob` legacy but it seems that some basic use cases depend on this feature.
